demisto_sdk/commands/common/content/objects/pack_objects/pre_process_rule/pre_process_rule.py

```diff
@@ -0,0 +1,45 @@
+from typing import Union
+
+import demisto_client
+from wcmatch.pathlib import Path
+
+from demisto_sdk.commands.common.constants import PRE_PROCESS_RULES, FileType
+from demisto_sdk.commands.common.content.objects.pack_objects.abstract_pack_objects.json_content_object import \
+    JSONContentObject
+
+PRE_PROCESS_RULES_PREFIX = 'preprocessrule-'
+
+
+class PreProcessRule(JSONContentObject):
+    def __init__(self, path: Union[Path, str]):
+        super().__init__(path, PRE_PROCESS_RULES)
+
+    def normalize_file_name(self) -> str:
+        """Add prefix to file name if not exists.
+
+        Examples:
+            1. "hello-world.json" -> "preprocessrule-hello-world.json"
+            2. "preprocessrule-hello-world.json" -> "preprocessrule-hello-world.json"
+
+        Returns:
+            str: Normalize file name.
+        """
+        output_basename = self._path.name
```

Comment:
seems quite similar to the overridden version in `general_object.py`, perhaps we could just set `self._prefix` and let `general_object` handle this?

Reply:
https://github.com/demisto/demisto-sdk/pull/1491/commits/429e145e24bcf90ad9bb1f13f5eb75e71a2e434b